### PR TITLE
Neocortex2

### PR DESCRIPTION
**Pull request by alialper**
In order to use the Arduino toolchain more correctly, we decided to define a new board in the Neocortex package and use the AVR_[board_name] #define to differentiate between board versions during compilation. Like in this Torsk commit:

https://github.com/CELLINKAB/Torsk/pull/230/commits/117b3d082f89aa394d8914348d19969a6de1e904

NOTE! Before merging, the circular reference to the branch in the package index json URL needs to be converted to "master"

### CODE REVIEW

File: neocortex/boards.txt [hunk 1]
```diff
@@ -25,3 +25,25 @@ neocortex.build.f_cpu=16000000L
...
+neocortex2.bootloader.high_fuses=0xD8
+neocortex2.bootloader.extended_fuses=0xFD
+neocortex2.bootloader.path=stk500v2
+neocortex2.bootloader.unlock_bits=0x3F
+neocortex2.bootloader.lock_bits=0x0F
+neocortex2.build.mcu=atmega2560
+neocortex2.build.f_cpu=16000000L
+neocortex2.build.board=AVR_NEOCORTEX2
+neocortex2.build.core=arduino:arduino
+neocortex2.build.variant=neocortex
```
Comment (platisd) [inline neocortex/boards.txt]: As discussed, can you add the second board as a menu item instead?

**Review by platisd — CHANGES_REQUESTED** [review]

**Comment by platisd** [comment]
Also a reminder, if we are to change the neocortex name let's not forget to change the [recipe that already uses it](https://github.com/CELLINKAB/meta-cellink/blob/master/recipes-platform/neocortex/neocortex-cellcyte_0.0.1.bb).

**Comment by alialper** [comment]
> Also a reminder, if we are to change the neocortex name let's not forget to change the [recipe that already uses it](https://github.com/CELLINKAB/meta-cellink/blob/master/recipes-platform/neocortex/neocortex-cellcyte_0.0.1.bb).

@platisd does the addition of Neocortex2 affect Cellcyte?

We also should adapt the recipes in meta-heartOS to pull in this repository and board package instead of using a pre-installed rambo board package.

**Comment by platisd** [comment]
> > Also a reminder, if we are to change the neocortex name let's not forget to change the [recipe that already uses it](https://github.com/CELLINKAB/meta-cellink/blob/master/recipes-platform/neocortex/neocortex-cellcyte_0.0.1.bb).
> 
> @platisd does the addition of Neocortex2 affect Cellcyte?
> 
> We also should adapt the recipes in meta-heartOS to pull in this repository and board package instead of using a pre-installed rambo board package.

I believe that if you don't change the existing naming the Cellcyte Marlin recipe will still work. It's not too difficult to find out if it broke and fix it though, so don't sweat over it too much. It was just mainly a note to myself that if the Cellcyte build suddenly breaks due to not finding a board it will be because of this. :)

File: README.md [hunk 2]
```diff
@@ -14,3 +15,47 @@ by adding a `,` before if more entries are already present.
...
+
+## Updating board packages
+
+Creating a new board package version involves updating the `./neocortex` directory as desired, creating a `.zip` archive from it with the correct version name, updating the `package_cellink_index.json` file, and pushing these changes to the desired branch of the Git repository.
+
+### Modifying the board package
+
+See the 3rd party Hardware specification that governs the board packages to be used to define boards:
+
+<https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5-3rd-party-Hardware-specification>
```
Comment (platisd) [inline README.md]: Use a link instead:
```
See the [3rd party Hardware specification](https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5-3rd-party-Hardware-specification) that governs the board packages to be used to define boards
```

File: README.md [hunk 3]
```diff
@@ -14,3 +15,49 @@ by adding a `,` before if more entries are already present.
...
+You can add different build targets as top level "boards", or alternatively as configurations of a single board. You can define such configuration options as "menu items" on top of the `boards.txt` file. These menu items will appear in the Arduino IDE "Tools" menu, under the board selection. These configurations can be used to override the board defaults. 
+
+1. Update the version number in `platform.txt`.
+
+### Packaging and Deployment
+
+1. Create a `.zip` archive from the `./neocortex` folder and name the archive `neocortex-X.X.X.zip`
+2. The `package_xxx_index.json` document is used by the Arduino IDE to install the board package, and needs to be updated. Update the `"version"` and the `"archiveFileName"`. The specification for the package index is described in the following document:
+
+    <https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.6.x-package_index.json-format-specification>
```
Comment (platisd) [inline README.md]: Use a link instead

**Review by platisd — CHANGES_REQUESTED** [review]
Looks good!

File: package_cellink_index.json [hunk 4]
```diff
@@ -12,12 +12,12 @@
...
                "name": "NeoCortex",
                "architecture": "avr",
-               "version": "1.0.0",
+               "version": "1.1.0",
                "category": "Contributed",
-               "url": "https://raw.githubusercontent.com/CELLINKAB/lowlevelboard-arduino-utils/master/neocortex-1.0.0.zip",
-               "archiveFileName": "neocortex-1.0.0.zip",
-               "checksum": "SHA-256:b93b0abb16515a9c0df408b2213ff854856c276689344331ce0cbd15e19ee916",
-               "size": "4155",
+               "url": "https://raw.githubusercontent.com/CELLINKAB/lowlevelboard-arduino-utils/neocortex2/neocortex-1.1.0.zip",
```
Comment (platisd) [inline package_cellink_index.json]: Looks good! Let's change that back to `master` and merge :+1:

**Review by platisd — CHANGES_REQUESTED** [review]

File: package_cellink_index.json (hunk 4 above)
Comment (alialper) [reply package_cellink_index.json]: thanks for reminding!

File: neocortex/boards.txt [hunk 5]
```diff
@@ -22,6 +21,15 @@ neocortex.bootloader.unlock_bits=0x3F
 neocortex.bootloader.lock_bits=0x0F
 neocortex.build.mcu=atmega2560
 neocortex.build.f_cpu=16000000L
-neocortex.build.board=AVR_NEOCORTEX
 neocortex.build.core=arduino:arduino
 neocortex.build.variant=neocortex
+
+########################################
+
+# default board is not enabled, and must be overridden by the revision menu:
```
Comment (platisd) [inline neocortex/boards.txt:29]: Shouldn't we enable the default board to be `2`?
Or did you discover a reason against this?

**Review by platisd — CHANGES_REQUESTED** [review]

**Pull request merged**